pkg/templateservicebroker/api/types.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// +genclient=true
+
+type TemplateProvision struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	Spec   TemplateProvisionSpec
+	Status TemplateProvisionStatus
+}
+
+type TemplateProvisionSpec struct {
+	Template  templateapi.Template
+	SecretRef kapi.LocalObjectReference
+}
+
+type TemplateProvisionStatus struct {
+	State TemplateProvisionState
```

Comment:
we should use Conditions which are the new phase/state mechanism.
see: 
https://github.com/openshift/origin/blob/master/pkg/deploy/api/v1/types.go#L298-L299